apps/snmp/snmp_stream.c

```diff
@@ -0,0 +1,475 @@
+/*
+ *
+  ***** BEGIN LICENSE BLOCK *****
+ 
+  Copyright (C) 2024 Olof Hagsand
+                2024 Mico Micic and Moser-Baer AG
+
+  This file is part of CLIXON.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  Alternatively, the contents of this file may be used under the terms of
+  the GNU General Public License Version 3 or later (the "GPL"),
+  in which case the provisions of the GPL are applicable instead
+  of those above. If you wish to allow use of your version of this file only
+  under the terms of the GPL, and not to allow others to
+  use your version of this file under the terms of Apache License version 2, 
+  indicate your decision by deleting the provisions above and replace them with
+  the  notice and other provisions required by the GPL. If you do not delete
+  the provisions above, a recipient may use your version of this file under
+  the terms of any one of the Apache License version 2 or the GPL.
+
+  ***** END LICENSE BLOCK *****
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "clixon_config.h" /* generated by config & autoconf */
+#endif
+
+#include <stdlib.h>
+#include <unistd.h>
+#include <errno.h>
+
+/* cligen */
+#include <cligen/cligen.h>
+
+/* clixon */
+#include <clixon/clixon.h>
+
+/* net-snmp */
+#include <net-snmp/net-snmp-config.h>
+#include <net-snmp/net-snmp-includes.h>
+#include <net-snmp/agent/net-snmp-agent-includes.h>
+
+#include "snmp_lib.h"
+
+/*
+ * SNMP v2 notification OID
+ */
+static oid notificationOid[] = {1, 3, 6, 1, 6, 3, 1, 1, 4, 1, 0};
+
+struct stream_socket{
+    qelem_t    sc_q;   /* queue header */
+    int        socket; /* socket */
+};
+/* Linked list of sockets used to listen for events
+ */
+static struct stream_socket *STREAM_SOCKETS = NULL;
+
+/*! Read smiv2:oid for given YANG node
+ *
+ * @param[in]  ys       YANG node
+ * @param[out] objid    oid
+ * @param[out] objidlen oid length
+ * @retval     0  OK
+ * @retval    -1  Error
+ */
+int
+get_oid_for_yang_node(yang_stmt *ys, oid *objid, size_t *objidlen)
```

Comment:
Style: please put each parameter on own line as described in 
https://github.com/clicon/clixon/blob/master/CONTRIBUTING.md#function-declarations
2: ... each parameter on own line
Also at all other places where formal parameters are declared.